etcdmgmt/etcd-mgmt.go

```diff
@@ -35,6 +36,11 @@ func StartEtcd() error {
 		"-initial-advertise-peer-urls", initialAdvPeerUrls,
 		"--initial-cluster", "default="+listenPeerUrls)
 
+	// Don't kill chlid process (etcd) upon ^C (SIGINT) of main glusterd process
```

Comment:
How about SIGKILL, SIGTERM? Aren't you aware that these signals are for shutting down a process?

Regarding running etcd instance from the code its straight forward that there is no check of whether a running etcd instance is available and if so etcd start shoild be skipped. When I pulled your changes I could easily see a defunct etcd process on a glusterd restart. I am not sure about what you have tested.

Moreover, this change will solve the issue of not letting the etcd instance go down on a glusterd restart but this doesn't cover the definition of daemonizing a process which should be a framework to manage multiple daemons. However I feel we can go ahead with it for now and later can come up with a generic framework.